Fixes a sporadic test problem with ruby.
See http://bugs.ruby-lang.org/issues/show/1714

The fix is from this revision, which does not appear in
the latest ruby 1.8.7-p374:
http://bugs.ruby-lang.org/projects/ruby-18/repository/revisions/24668

--- ruby-1.8.7-p374-orig/lib/test/unit/collector/dir.rb	2007-02-12 15:01:19.000000000 -0800
+++ ruby-1.8.7-p374/lib/test/unit/collector/dir.rb	2013-12-05 17:46:31.105788000 -0800
@@ -91,7 +91,7 @@
           end
           find_test_cases(already_gathered).each{|t| add_suite(suites, t.suite)}
         ensure
-          $:.delete_at($:.rindex(dir)) if(dir)
+          $:.delete_at($:.index(dir)) if(dir)
         end
 
 	def realdir(path)
